File: Army(c++)/State/SoldierState.cpp
#include "SoldierState.hpp"
#include "../Basic/Unit.hpp"

SoldierState::SoldierState(Unit* thisUnit) : thisUnit(thisUnit) {}

void SoldierState::changeUnit(Unit* enemy) {
    std::cout << "No ability!!!"  << std::endl;
}

void SoldierState::transformTo() {
    std::cout << "No ability!!!"  << std::endl;
}

File: Army(java)/State/SoldierState.java
package State;
import Basic.*;

public class SoldierState implements State {
    protected Unit thisUnit;

    public SoldierState(Unit unit) {
        this.thisUnit = unit;
    }

    public void printState() {
        System.out.println("\tI'm Soldier!!!");
    }

    public void changeUnit(Unit enemy) throws UnitIsDeadException {
        System.out.println("No ability!!!");
    }

    public void transformTo() throws UnitIsDeadException {
        System.out.println("No ability!!!");
    }
}
